src/fabric.c

```diff
@@ -669,6 +675,50 @@ static void ofi_find_prov_libs(void)
 		free(lib);
 	}
 }
+
+static void ofi_load_dl_prov(void)
+{
+	char **dirs;
```

Comment:
I got a warning when compiling this with gcc on rhel 8.1: 
```
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.  -I./include -D_GNU_SOURCE -D__USE_XOPEN2K8 -DSYSCONFDIR=\"/usr/local/etc\" -DRDMADIR=\"@rdmadir@\" -DPROVDLDIR=\"/usr/local/lib/libfabric\" -I./prov/sockets/include -I./prov/sockets  -I./prov/verbs/src/ep_rdm -I./prov/verbs/src/ep_dgram   -D__LIBUSNIC__ -DWANT_DEBUG_MSGS=0 -DHAVE_LIBNL3=1 -I/usr/include/libnl3 -I./prov/usnic/src/usnic_direct   -I./prov/hook/include -I./prov/hook/perf/include -I./prov/hook/hook_debug/include  -Wall -O2 -DNDEBUG -fvisibility=hidden -Wall -Wundef -Wpointer-arith -MT src/libfabric_la-fabric.lo -MD -MP -MF src/.deps/libfabric_la-fabric.Tpo -c -o src/libfabric_la-fabric.lo `test -f 'src/fabric.c' || echo './'`src/fabric.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./include -D_GNU_SOURCE -D__USE_XOPEN2K8 -DSYSCONFDIR=\"/usr/local/etc\" -DRDMADIR=\"@rdmadir@\" -DPROVDLDIR=\"/usr/local/lib/libfabric\" -I./prov/sockets/include -I./prov/sockets -I./prov/verbs/src/ep_rdm -I./prov/verbs/src/ep_dgram -D__LIBUSNIC__ -DWANT_DEBUG_MSGS=0 -DHAVE_LIBNL3=1 -I/usr/include/libnl3 -I./prov/usnic/src/usnic_direct -I./prov/hook/include -I./prov/hook/perf/include -I./prov/hook/hook_debug/include -Wall -O2 -DNDEBUG -fvisibility=hidden -Wall -Wundef -Wpointer-arith -MT src/libfabric_la-fabric.lo -MD -MP -MF src/.deps/libfabric_la-fabric.Tpo -c src/fabric.c  -fPIC -DPIC -o src/.libs/libfabric_la-fabric.o
src/fabric.c: In function 'fi_ini':
src/fabric.c:712:3: warning: 'dirs' may be used uninitialized in this function [-Wmaybe-uninitialized]
   ofi_free_string_array(dirs);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/fabric.c:681:9: note: 'dirs' was declared here
  char **dirs;
         ^~~~
```
Not sure why, ofi_str.h is in the include list and at top of file.

Reply:
fixed - added missing else check.  Also updated to support '@' by itself

Reply:
Looks good to me